client/src/components/RelatedProducts/aRelatedProduct.jsx

```diff
@@ -59,18 +64,37 @@ class ARelatedProduct extends React.Component {
     });
   }
 
+  openModal() {
+    this.setState(
+      { showModal: !this.state.showModal },
+    );
+  }
+
   render() {
     const { name } = this.state;
     const { price } = this.state;
     // const { description } = this.state;
     const { averageRating } = this.state;
     const { photos } = this.state;
     const { category } = this.state;
+    const { showModal } = this.state;
```

Comment:
As the amount of state variables grow, combining the destructuring statements could save a couple lines of code, i.e. const { name, price, ..., lastState } = this.state